### PR TITLE
Handle stop contact channel events

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/channels/models.py [hunk 1]
```diff
@@ -1492,6 +1495,13 @@ def handle(self):
         elif self.event_type == ChannelEvent.TYPE_FOLLOW:
             handled = Trigger.catch_triggers(self, Trigger.TYPE_FOLLOW, self.channel)
 
+        elif self.event_type == ChannelEvent.TYPE_STOP_CONTACT:
+            user = get_anonymous_user()
+            contact = Contact.get_or_create(self.org, user, name=None, urns=[self.contact_urn.urn],
+                                            channel=self.channel)
+            contact.stop(user)
+            handled = bool(contact)
```
Comment (nicpottier) [inline temba/channels/models.py]: why the bool?

Comment (norkans7) [reply temba/channels/models.py]: not needed really I had borrowed that from catch_triggers method but here we are sure we 'll always have a contact

File: temba/channels/migrations/0084_auto_20180115_1606.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+    dependencies = [
+        ('channels', '0083_use_secret_config'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='channelevent',
+            name='event_type',
+            field=models.CharField(choices=[('unknown', 'Unknown Call Type'), ('mt_call', 'Outgoing Call'), ('mt_miss', 'Missed Outgoing Call'), ('mo_call', 'Incoming Call'), ('mo_miss', 'Missed Incoming Call'), ('stop_contact', 'Stop Contact'), ('new_conversation', 'New Conversation'), ('referral', 'Referral'), ('follow', 'Follow')], help_text='The type of event', max_length=16, verbose_name='Event Type'),
```
Comment (rowanseymour) [inline temba/channels/migrations/0084_auto_20180115_1606.py]: side note - but is it worth maintaining these choice lists? They're not used unless you were to create a form for creating channelevents

Comment (norkans7) [reply temba/channels/migrations/0084_auto_20180115_1606.py]: the builder was failing because of the missing migration then I added it

Comment (rowanseymour) [reply temba/channels/migrations/0084_auto_20180115_1606.py]: Sure - I'm wondering about removing the `choices` stuff altogether on models where we don't use it with a form

Comment (nicpottier) [reply temba/channels/migrations/0084_auto_20180115_1606.py]: It isn't only when it is a form is it? also if we ever say something like `event_type_display` in a template, which in this case may actually occur? (though we'd have to localize them)

I'm fine removing them.

Comment (norkans7) [reply temba/channels/migrations/0084_auto_20180115_1606.py]: I think that is fine

**Pull request merged**